cache/redis_cache.go

```diff
@@ -334,7 +375,7 @@ func (r *redisStreamReader) Read(destBuf []byte) (n int, err error) {
 
 		// others errors, such as timeouts
 		if err != nil && !errors.Is(err, redis.Nil) {
-			log.Errorf("failed to get key %s with error: %s", r.key, err)
+			log.Debugf("failed to get key %s with error: %s", r.key, err)
 			return bytesWritten, err
```

Comment:
Can we also propagate `RedisCacheError` here with the context representing a timeout? This way we could eliminate track partial reads failures. But IMO if this can happen, and code suggest it can, we have an issue with the read design because it results with failure reported to client.

Reply:
When I wrote this part I thought this could happen if redis is unresponsive/down/unreachable during a stream (despite all the retries in the redis client).
It should be a very edge case but if it happens there is nothing we can do: all the queries that were currently streamed will fail  (only the queries taking more than 100k Bytes are streamed)/ All the ones that just started the stream will succeed. IMHO, the probability should be the same as having a chproxy node down or a clickhouse shard down which will result in the same issue.

I'll modify this part to throw a RedisCacheError with the root error to keep the context.